src/autogluon/bench/eval/benchmark_context/output_context.py

```diff
@@ -208,6 +216,24 @@ def get_single_leaderboard(self, columns_to_keep, with_infer_speed, i, num_conte
             print(f"SUCCESS: {print_msg}")
             return combined_full
 
+    def get_model_failures(self) -> pd.DataFrame | None:
+        """
+        Load and return the model failures CSV as a pandas DataFrame if it exists, else return None.
+
+        Will merge with the results to get additional information, akin to the leaderboard output.
+        """
+        results = self.load_results()
+        try:
+            model_failures_df = self.load_model_failures()
+        except Exception as e:
+            print(f"FAILURE:\n" f"\t{e.__class__.__name__}: {e}")
+            return None
+        else:
+            results = results.rename(columns={"framework": "framework_parent"})
+            model_failures_df["id"] = results["id"][0]
+            model_failures_full_df = pd.merge(model_failures_df, results, on="id", how="left")
```

Comment:
Does failed model also have an entry in the `results`? I'm curious on how the two look differently.

Reply:
Failed models will have an entry in `results` for the overall framework run, but will not be present in the `leaderboard` artifact, since only successful models are part of `leaderboard`.